app/team.go

```diff
@@ -1014,6 +1014,19 @@ func (a *App) SoftDeleteTeam(teamId string) *model.AppError {
 	return nil
 }
 
+func (a *App) RestoreTeam(teamId string) *model.AppError {
+	team, err := a.GetTeam(teamId)
+	if err != nil {
+		return err
+	}
+	result := <-a.Srv.Store.Team().SetDeleteAt(team.Id, 0, model.GetMillis())
```

Comment:
I think here we must follow the same approach of the previous method, set the deleteat to 0 and use the update store method. This way we can remove the store modifications.

Reply:
Will update the PR @jespino